app/services/atmosphere/optimizer.rb

```diff
@@ -14,6 +14,12 @@ def run(hint)
 
     #private
     def satisfy_appliance(appliance)
+
+      action = Action.create { |action|
+        action.appliance = appliance
+        action.action_type = :satisfy_appliance
+      }
```

Comment:
Here and in scaling you are creating only `action`, no `action_logs` are added. Maybe information when scaling started/finished, what is the results (e.g. id of VMs started/reused) should be logged here.

The same comment is valid for scale method.